foundation/plugins/org.jboss.tools.foundation.ui/src/org/jboss/tools/foundation/ui/util/BrowserUtility.java

```diff
@@ -66,4 +104,280 @@ public void openUrl(String url, IWebBrowser browser, String pluginId, ILog log)
 			log.log(errorStatus);
 		}
 	}
+	
+	static public void openExtenalBrowser(String URL) {
+		checkedCreateExternalBrowser(URL,FoundationUIPlugin.PLUGIN_ID,FoundationUIPlugin.getDefault().getLog());
+	}
+	
+	/**
+	 * Creates an SWT Browser
+	 * @param style
+	 * @param parent
+	 * @return
+	 */
+	
+	public static Browser createBrowser(int style, Composite parent) {
+		return createBrowser(style, parent, getPreferredBrowser());
+	}
+
+	/**
+	 * Creates an SWT Browser
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 */
+	public static Browser createBrowser(Composite parent) {
+		return createBrowser(SWT.READ_ONLY | SWT.NO_SCROLL, parent, getPreferredBrowser());
+	}
+	
+	/**
+	 * Creates an SWT Browser
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 */
+	public static Browser createBrowser(Composite parent, int preferredBrowser) {
+		return createBrowser(SWT.READ_ONLY | SWT.NO_SCROLL, parent, preferredBrowser);
+	}
+
+	/**
+	 * Creates an SWT Browser. Returns null if can't create a browser.
+	 * @param style
+	 * @param parent
+	 * @param preferredBrowser SWT.MOZILLA, SWT.WEBKIT or SWT.NONE
+	 * @return
+	 * @see org.eclipse.swt.browser.Browser
+	 */
+	public static Browser createBrowser(int style, Composite parent, int preferredBrowser) {
+		/*
+			// We can provide webkit in this way:
+			   String defaultBrowser = System.getProperty(Browser.PROPERTY_DEFAULTTYPE);
+			   boolean hasDefaultBrowser = defaultBrowser != null;
+			   System.getProperties().setProperty(Browser.PROPERTY_DEFAULTTYPE, "webkit");
+		*/
+		if(System.getProperty("skipBrowserCreation")!=null) {
```

Comment:
This was a change by Denis. I suspect he had a reason to do it.   I will say, though, that the presence of the property has allowed me to test the behavior better and fix a bunch of bugs I would not have been able to replicate otherwise.